starky/src/stark.rs

```diff
@@ -96,4 +97,36 @@ pub trait Stark<F: RichField + Extendable<D>, const D: usize>: Sync {
             batches: vec![zeta_batch],
         }
     }
+
+    /// Pairs of lists of columns that should be permutations of one another. A permutation argument
+    /// will be used for each such pair. Empty by default.
+    fn permutation_pairs(&self) -> Vec<PermutationPair> {
+        vec![]
+    }
+}
+
+/// A pair of lists of columns, `lhs` and `rhs`, that should be permutations of one another.
+/// In particular, there should exist some permutation `pi` such that for any `i`,
+/// `lhs[i] = rhs[pi(i)]`.
```

Comment:
Hmm I'm not sure I understand this. From this comment, it seems like `lhs` is a set of column indices, and `rhs` is a permutation of `lhs`. What information does that give us about permutations of columns?

Reply:
Oh yeah I wrote this wrong. Does it make sense now?

Reply:
Yeah makes sense! So the permutation `pi` would be the same for all pairs in `PermutationPair`? Where would we use a `PermutationPair` with lists of length `>1`?

Reply:
Yeah exactly. I think we'll need that property for the memory argument, where we'll have timestamp, address, and value columns, plus a sorted version of the same columns. I.e. we'd want the same permutation to map each unsorted col to each sorted col.

Reply:
Oh yeah that sounds good!